src/navigation/gate.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+from typing import ClassVar, Optional
+from unicodedata import normalize
+from context import Gate
+
+import numpy as np
+
+from context import Context, Environment
+from state import BaseState
+from trajectory import Trajectory
+from dataclasses import dataclass
+from drive import get_drive_command
+from util.np_utils import normalized, perpendicular_2d
+
+STOP_THRESH = 0.2
```

Comment:
Make these dynamic parameters

Reply:
so rn we do this in p much every state. I do generally agree tho so maybe would be a good issue for a new member to fix?